src/main/java/com/example/Library/services/BookService.java

```diff
@@ -74,15 +76,17 @@ public Optional<Book> returnBook(long id) {
                 });
     }
 
-    public Iterable<Book> getAllBooksWithFilterByTitle(String title) {
+    public Page<Book> getAllBooksWithFilterByTitle(Optional<Integer> page, Optional<String> sortBy, String title) {
+
         if (title != null) {
             ArrayList<Book> books = (ArrayList<Book>) bookRepository.getBooksByTitle(title);
-            if (books.size()==0){
+            if (books.size() == 0) {
                 throw new BookNotFoundByTitleException(title);
-            }else {
-                return books;
+            } else {
+                return new PageImpl<>(books);
             }
         }
-        return bookRepository.findAll();
+        return bookRepository.findAll(PageRequest.of(page.orElse(0), 5,
```

Comment:
pls try to avoid using 'magic number' such as '5', you can create a constant with a meaningful name in this class, and make use of it here
([see magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)#:~:text=The%20term%20magic%20number%20or,1%20manuals%20of%20the%201960s.))